onnxruntime/core/framework/session_state_initializer.cc

```diff
@@ -243,28 +245,35 @@ common::Status SaveInitializedTensors(const Env& env, const std::basic_string<PA
 
   //1. first plan the memory
   const onnxruntime::InitializedTensorSet& initialized_tensor_set = graph.GetAllInitializedTensors();
-  std::unordered_map<int, const ONNX_NAMESPACE::TensorProto*> id_to_initialized_tensor;
+  // map of ort value index to <initializer name, TensorProto>
+  using NameAndProto = std::pair<const std::string, const ONNX_NAMESPACE::TensorProto*>;
```

Comment:
>TensorProto [](start = 74, length = 11)

TensorProto has names for initializers, why do we need another name? Could they be different?

Reply:
I think they should be the same, but the existing handling of the TensorProto name being empty made me assume that wasn't the case. i.e. this code
    const char* name = entry.second->has_name() ? entry.second->name().c_str() : "";

As the input to this is initialized_tensor_set where the name is the key, I preferred that value. 

Now that I think about it more, I'm not sure what a TensorProto with an empty name could be used for, so I think this could be simplified to use the name from TensorProto and ignore or error out on any TensorProto where the name is empty. Does that seem reasonable?

---
In reply to: [299589384](https://github.com/microsoft/onnxruntime/pull/1324#discussion_r299589384) [](ancestors = 299589384)

Reply:
Sounds good to me.

---
In reply to: [299709883](https://github.com/microsoft/onnxruntime/pull/1324#discussion_r299709883) [](ancestors = 299709883,299589384)